Assets/MRTK/SDK/Features/UX/Scripts/StateVisualizer/AnimationTarget.cs

```diff
@@ -0,0 +1,195 @@
+            var animatablePropertyTypes = TypeCacheUtility.GetSubClasses<StateAnimatableProperty>();
+            Type animatablePropertyType = animatablePropertyTypes.Find((type) => type.Name.StartsWith(animatablePropertyTypeName));
+
+            if (animatablePropertyType != null)
+            {
+
+                if (CanAddAnimatableProperty(animatablePropertyTypeName))
+                {
+                    // If a state has an associated event configuration class, then create an instance with the matching type
+                    animatableProperty = Activator.CreateInstance(animatablePropertyType) as StateAnimatableProperty;
+
+                }
+                else
+                {
+                    animatableProperty = null;
+                    Debug.LogError($"Only one {animatablePropertyTypeName} animatable property can be present for this target object.");
+                }
+
+                if (animatableProperty != null)
+                {
+                    animatableProperty.StateName = stateName;
+                    animatableProperty.Target = Target;
+
+                    // Generate unique id for shader properties because multiple shader properties can be 
+                    // animated on a single target object
+                    if (animatablePropertyTypeName.Contains("Shader"))
+                    {
+                        int shaderPropertyID = GenerateIDShaderProperty();
+
+                        animatableProperty.AnimatablePropertyName = animatablePropertyTypeName + "_" + shaderPropertyID;
+                    }
+
+                    StateAnimatableProperties.Add(animatableProperty);
+                }
+            }
+            else
+            {
+                animatableProperty = null;
+                Debug.Log("The animatableProperty property name given does not have a matching configuration type");
+            }
+
+            return animatableProperty;
+        }
+
+        public StateAnimatableProperty AddNewAnimatableProperty(AnimatableProperty animatablePropertyTypeName, string stateName)
+        {
+            return CreateAnimatablePropertyInstance(animatablePropertyTypeName.ToString(), stateName);
+        }
+
+        private bool CanAddAnimatableProperty(string animatablePropertyTypeName)
+        {
+            // Multiple animatable shader properties can be present on one target object
+            if (animatablePropertyTypeName.Contains("Shader"))
+            {
+                return true;
+            }
+
+            // Ensure that there is only one Scale and Position Offset property per target game object
+            foreach (var animatableProp in StateAnimatableProperties)
+            {
+                if (animatableProp.AnimatablePropertyName == animatablePropertyTypeName)
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private int GenerateIDShaderProperty()
```

Comment:
Super minor, more as an FYI. I think you can use the LINQ [.Count()](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.count?view=net-5.0) function to do the same thin. Something like `StateAnimatableProperties.Count((x => x is ShaderStateAnimatableProperty))`